File: cpp/1015. 可被 K 整除的最小整数.cpp
class Solution {
public:
    int smallestRepunitDivByK(int k) {
        unordered_map<int,bool> hashMap;
        if(k%2==0||k%5==0){
            return -1;
        }
        int len=1;
        int resid=1%k;
        while(resid!=0){
            resid=(resid*10+1)%k;
            len++;
            if(hashMap[resid]){
                return -1;
            }
           
            hashMap[resid]=true;
        }
        return len;
    }
};

File: golang/1015. 可被 K 整除的最小整数.go
func smallestRepunitDivByK(k int) int {
	// 如果 k 是偶数或者以 5 结尾，那么一定不存在这样的正整数，直接返回 -1。
	if k%2 == 0 || k%5 == 0 {
		return -1
	}
	// cur 代表当前的余数，初始值为 0，res 代表数字 1 的个数，初始值为 1。
	cur, res := 0, 1

	// 不断计算下一个数字的余数，直到找到一个可行的解或者发现不存在可行的解。
	for {
		cur = (10*cur + 1) % k
		if cur == 0 {
			return res
		}
		res++
	}
}